skrifa/src/scale/mod.rs

```diff
@@ -1,13 +1,128 @@
-/*!
-Glyph loading and scaling.
-*/
+//! Extraction of glyph outlines.
+//!
+//! Scaling is the process of decoding an outline, applying variation deltas,
+//! and executing [hinting](https://en.wikipedia.org/wiki/Font_hinting) instructions
+//! for a glyph of a particular size.
+//!
+//! The process starts with construction of a [`Context`] that can be used to build
+//! and configure a [`Scaler`] which can generate scaled paths for glyphs.
+//!
+//! Read on for more detail.
+//!
+//! ## Building a scaler
+//!
+//! Scaling requires temporary memory allocations and, in the case of hinting,
+//! also benefits from caching the state of the hinting setup programs. To amortize the
+//! cost of heap allocations and support caching, the [`Context`] type is provided.
+//! This type is opaque and offers a single method called [`new_scaler`](Context::new_scaler)
+//! that produces a [`ScalerBuilder`] to configure and builder a [`Scaler`].
+//!
+//! Assuming you have some `font` (any type that implements [`TableProvider`](read_fonts::TableProvider)),
+//! this will build a scaler for a size of 16px:
+//!
+//! ```
+//! # use skrifa::{scale::*, Size};
+//! # fn build_scaler(font: read_fonts::FontRef) {
+//! let mut context = Context::new();
+//! let mut scaler = context.new_scaler()
+//!     .size(Size::new(16.0))
+//!     .build(&font);
+//! # }
+//! ```
+//!
+//! For variable fonts, the [`variation_settings`](ScalerBuilder::variation_settings) method can
+//! be used to specify user coordinates for selecting an instance:
+//!
+//! ```
+//! # use skrifa::{scale::*, Size};
+//! # fn build_scaler(font: read_fonts::FontRef) {
+//! let mut context = Context::new();
+//! let mut scaler = context.new_scaler()
+//!     .size(Size::new(16.0))
+//!     .variation_settings(&[("wght", 720.0), ("wdth", 75.0)])
```

Comment:
not related to the PR but the question that pops into my head, here is, "what if the axis doesn't exist? is there an error condition?

Reply:
As mentioned in the second paragraph here: https://github.com/googlefonts/fontations/pull/340#discussion_r1160117009, doing this sort of validation provides very little value. Every production text layout engine that I'm aware of will simply drop unsupported axes (and clamp out of range values) so I chose to support that common case here by default. For clients that want to be more strict about their inputs, they can use the forthcoming variations metadata API (or the associated underlying read-fonts code) to do so. I can provide a `validate_settings` method when I do the variations PR to make that simpler.